docusaurus/docs/extensions/extensions-configuration.md

```diff
@@ -0,0 +1,956 @@
+
+```ts
+<-YOUR-RANCHER-INSTANCE-BASE-URL->/c/_/fleet
+```
+
+In terms of the route definition (Vue Router), we would translate this url to:
+
+```ts
+const clusterManagerRoute = {
+  name: 'c-cluster-product',
+  path: 'c/:cluster/:product',
+  params: {
+    cluster: '_',
+    product: 'fleet'
+  }
+}
+```
+
+As we can see from the example above, we have defined on the `path` the wildcards for `cluster` and `product`. Also we can see the definition of `params` property, which is needed for internal app navigation and where we define the `cluster` value as `_` , which in terms of the app context this means that we are using a "blank cluster" which translates that the app doesn't need to care about the cluster context for the Fleet product to run. Also we are defining `product` value as `fleet`, which in turn tells the app  what is the correct product to load.
+
+With this pattern of wildcards and `params` in mind, then how does the route structure should look like for a top-level Extension product? In short, it needs to follow this pattern:
+
+```ts
+const YOUR_EXT_PRODUCT_NAME = 'myExtension';
+
+const baseRouteForATopLevelProduct = {
+  name: `${ YOUR_EXT_PRODUCT_NAME }-c-cluster`,
+  path: `/${ YOUR_EXT_PRODUCT_NAME }/c/:cluster`,
+  params: {
+    cluster: '_',
+    product: YOUR_EXT_PRODUCT_NAME
+  }
+}
+```
+
+As we can see we have dismissed the `product` wildcard on the `path` and replaced it with the Extension product name to make it unique. With the `product` param we make sure that the is taken to the correct product at all time.
+This structure on the above example ensures that all the wiring needed for the Extension to work properly on Rancher Dashboard is done. There's even the case where the wildcard `resource` needs to be defined in order to display information about Kubernetes resources or custom CRDs. An example of a resource route in a top-level Extension product would be:
+
+```ts
+const YOUR_EXT_PRODUCT_NAME = 'myExtension';
+const RESOURCE_NAME = 'my-resource-name';
+
+const routeForATopLevelProductResource = {
+  name: `${ YOUR_EXT_PRODUCT_NAME }-c-cluster-resource`,
+  path: `/${ YOUR_EXT_PRODUCT_NAME }/c/:cluster/:resource`,
+  params: {
+    cluster: '_',
+    product: YOUR_EXT_PRODUCT_NAME
+    resource: RESOURCE_NAME
+  }
+}
+```
+
+With this quick guide on routing for top-level Extension products, we should be ready to cover the registration of custom pages, resource pages and general route definition. 
+
+
+## Defining a custom page for an Extension (virtualType)
+As we've seen from the previous example, a developer can register a top-level product with the `product` function. How about adding a custom page to your extension product? To do that, we can use the function `virtualType` coming from `$plugin.DSL`. As an example usage of that method, one could do the following:
+
+```ts
+// this is the definition of a "blank cluster" for Rancher Dashboard
+import { BLANK_CLUSTER } from '@shell/store';
```

Comment:
I have tried this on the new create/app repo and I don't get any problems running the scripts `build-pkg` or `publish-pkgs`. We can get together to check this out.
Nevertheless I think it's important to explain to the user the notion of a `BLANK_CLUSTER` and, imo, I would prefer to see them use our constant rather than `_` because if we do change something on this part we might have a change to mitigate breaking changes.

We can chat about this if you want 🙏